File: app/src/test/java/com/benoitquenaudon/tvfoot/red/app/injection/module/TestSchedulerModule.java
package com.benoitquenaudon.tvfoot.red.app.injection.module;

import dagger.Module;
import dagger.Provides;
import com.benoitquenaudon.tvfoot.red.app.common.schedulers.BaseSchedulerProvider;
import com.benoitquenaudon.tvfoot.red.app.common.schedulers.ImmediateSchedulerProvider;
import javax.inject.Singleton;

@Module public class TestSchedulerModule {

  @Provides @Singleton BaseSchedulerProvider provideSchedulerProvider() {
    return new ImmediateSchedulerProvider();
  }
}

File: app/src/test/java/com/benoitquenaudon/tvfoot/red/injection/module/TestSchedulerModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import com.benoitquenaudon.tvfoot.red.app.common.schedulers.BaseSchedulerProvider
import com.benoitquenaudon.tvfoot.red.app.common.schedulers.ImmediateSchedulerProvider
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module object TestSchedulerModule {

  @JvmStatic @Provides @Singleton fun provideSchedulerProvider(): BaseSchedulerProvider {
    return ImmediateSchedulerProvider()
  }
}
